src/sysrepo.c

```diff
@@ -352,8 +352,17 @@ sr_connect(const sr_conn_options_t opts, sr_conn_ctx_t **conn_p)
         sr_conn_free(conn);
         if (created) {
             /* remove any created SHM so it is not considered properly created */
-            shm_unlink(SR_MAIN_SHM);
-            shm_unlink(SR_EXT_SHM);
+            char *shm_name = NULL;
+            if (sr_path_main_shm(&shm_name) == NULL) {
```

Comment:
Please use `!` instead of `== NULL` just like in the rest of the code.

Reply:
done